archunit-junit/junit5/engine/src/test/java/com/tngtech/archunit/junit/ArchUnitTestEngineTest.java

```diff
@@ -1122,6 +1116,61 @@ void with_reason() {
         }
     }
 
+    @Nested
+    class GeneratesDisplayName {
+        @Test
+        void for_field_by_replacing_underscores_with_blanks_if_property_is_set_to_true() {
+            EngineDiscoveryTestRequest discoveryRequest = new EngineDiscoveryTestRequest().withClass(SimpleRuleField.class);
+
+            ArchConfiguration.get().setProperty(DisplayNameResolver.DISPLAYNAME_REPLACE_UNDERSCORES_PROPERTY_NAME, "true");
```

Comment:
If we do this we should also have some

```
@AfterEach
void resetConfiguration() {
    ArchConfiguration.get().reset();
}
```

Otherwise other tests might start to behave surprisingly / flaky :wink: